src/libraries/System.Private.CoreLib/src/System/Environment.OSX.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using nint = System.IntPtr;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static OperatingSystem GetOSVersion()
+        {
+            int major = 0;
+            int minor = 0;
+            int patch = 0;
+
+            IntPtr processInfo = objc_msgSend(objc_getClass("NSProcessInfo"), sel_getUid("processInfo"));
+            if (processInfo != IntPtr.Zero)
+            {
+                NSOperatingSystemVersion osVersion = get_operatingSystemVersion(processInfo, sel_getUid("operatingSystemVersion"));
+                major = osVersion.majorVersion.ToInt32();
+                minor = osVersion.minorVersion.ToInt32();
+                patch = osVersion.patchVersion.ToInt32();
+            }
+
+            // For compatibility reasons with Mono, PlatformID.Unix is returned on MacOSX. PlatformID.MacOSX
+            // is hidden from the editor and shouldn't be used.
+            return new OperatingSystem(PlatformID.Unix, new Version(major, minor, patch));
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct NSOperatingSystemVersion
+        {
+            public nint majorVersion;
+            public nint minorVersion;
+            public nint patchVersion;
+        }
+
+        [DllImport("libobjc.dylib")]
```

Comment:
Is this is the right path to use?

The non-absolute paths tend to break on OSX when people happen to run with custom `DYLD_FALLBACK_LIBRARY_PATH`. See https://github.com/dotnet/runtime/issues/24095 for example

Reply:
This is what we were using for System.Drawing:

https://github.com/dotnet/runtime/blob/f5ecd6bc936cb9502d5faf8e156d00cf526c9b76/src/libraries/System.Drawing.Common/src/System/Drawing/macFunctions.cs#L211-L222

Reply:
Does it work when you do `export DYLD_FALLBACK_LIBRARY_PATH=/nothing` ?

Reply:
We (the Xamarin.iOS/Mac team) always use full paths, because we've run into cases where it doesn't work.

I don't remember anymore exactly what happened, but we still have this in our code base: https://github.com/xamarin/xamarin-macios/blob/2926083d205fa47c402fa19397d2444d66dfaa2a/src/ObjCRuntime/Dlfcn.cs#L114-L126

It could have been something Apple broke in a beta (that code was originally committed on June 10, 2015, in the middle of WWDC/just after Apple released the first iOS 9 beta), and then fixed later on, but since then we've always used full paths, just in case Apple does something strange again.

Reply:
For reference, this is what Interop.Libraries.cs currently contains:
https://github.com/dotnet/runtime/blob/3b68eb3b1edc5a0ed0f4758668937fe4de17c431/src/libraries/Common/src/Interop/OSX/Interop.Libraries.cs#L5-L18

Reply:
I've added the full path: `/usr/lib/libobjc.dylib`